src/shared/utils.ts

```diff
@@ -1271,3 +1384,56 @@ export function pushNotNull(array: any[], new_item?: any) {
     array.push(...new_item);
   }
 }
+
+/**
+ * Converts an undefined or null value to an Option
+ *
+ * Note: You cannot use a prototype, because it won't be able to handle null values
+ */
+export function toOption<T>(val: T): Option<T> {
```

Comment:
This is the main change, converting nulls and undefineds to rust-style options.

Because typescript doesn't have first class support for these monads, I have to do:

Receive API with nulls <-> Lemmy UI with Monads <--> Send API with nulls.

Even using the Options internally helped me check and fix several bugs during this process, and clean up a LOT of code.

Reply:
You could consider adding this library to lemmy-js-client as well.

Reply:
It'd require something like this, and might be a lot of work / boilerplate to add: https://github.com/JohnWeisz/TypedJSON

Reply:
That library didn't work, but I found a way to do this using this library: https://github.com/typestack/class-transformer

Adds a good amount of boilerplate, but it fully works deserializing and serializing.